airbyte-webapp-e2e-tests/cypress/pages/sourcePage.ts

```diff
@@ -1,17 +1,20 @@
-const newSource = "button[data-id='new-source'";
+const newSource = "button[data-id='new-source']";
 
 export const goToSourcePage = () => {
   cy.intercept("/api/v1/sources/list").as("getSourcesList");
   cy.visit("/source");
   cy.wait(3000);
-  cy.wait("@getSourcesList");
 };
 
 export const openSourceDestinationFromGrid = (value: string) => {
   cy.get("div").contains(value).click();
 };
 
 export const openNewSourceForm = () => {
-  cy.get(newSource).click();
+  cy.wait("@getSourcesList").then(({ response }) => {
```

Comment:
Since we won’t have a button in the empty state (no sources set up yet) we need to make a decision based on the response on this request.